File: solutions/Virtual-Assistant/src/csharp/Microsoft.Bot.Solutions/Extensions/TurnContextEx.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.Bot.Builder;
using Microsoft.Bot.Solutions.Middleware;

namespace Microsoft.Bot.Solutions.Extensions
{
    public static class TurnContextEx
    {
        public static (string intent, double score)? GetTopIntent(this ITurnContext context)
        {
            return context.TurnState.Get<RecognizerResult>(LuisRecognizerMiddleware.LuisRecognizerResultKey)?.GetTopScoringIntent();
        }
    }
}

File: templates/typescript/samples/sample-skill/src/extensions/turnContextEx.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { TurnContext } from 'botbuilder';
import { SkillValidation, ClaimsIdentity } from 'botframework-connector';

export namespace TurnContextEx {

    export function isSkill(turnContext: TurnContext): boolean {
        const botIdentity = turnContext.turnState.get('BotIdentity');
        return botIdentity instanceof ClaimsIdentity && SkillValidation.isSkillClaim(botIdentity.claims) ? true : false;
    }

}